test/powershell/Modules/Microsoft.PowerShell.Utility/WebCmdlets.Tests.ps1

```diff
@@ -366,6 +366,14 @@ function GetMultipartBody {
     return , $multipartContent
 }
 
+filter FilterVerboseMessages ([string]$Message) {
+    if ($_ -and $_ -is [System.Management.Automation.VerboseRecord]) {
```

Comment:
I looked into the .Streams.Verbose and had a lot of trouble with it, but have something that works now. There was no shared code, but plenty of places with examples of how it was done.

I think my use case is a bit different because I am wanting to run multiple commands in the pipeline with state between each and then check whether a warning is raised each time. I have to move the state in and out of the created PowerShell session. 

In the context for persistence checks I have this function:
```powershell

    $connWarning = [regex]::new('WebSession.+HTTP')
    function RunCheckingPersistence {
            param(
                [uri]$Uri,
                [string]$Command,
                [object]$Session,
                [switch]$LostPersistence,
                [switch]$CaptureSession
            )

            $pwsh = [PowerShell]::Create()
            $pwsh.Runspace.SessionStateProxy.SetVariable('uri', $Uri)
            if ($Session) {
                $pwsh.Runspace.SessionStateProxy.SetVariable('Session', $Session)
                $command = "$command -WebSession `$Session"
            }
            if ($CaptureSession) {
                $command = "$command -SessionVariable Session"
            }
            $script = "`$null = $command -Verbose"
            $pwsh.AddScript($script).Invoke()
            $session = $pwsh.Runspace.SessionStateProxy.GetVariable('Session')

            $expectedWarnings = if ($LostPersistence) { 1 } else { 0 }
            ($pwsh.Streams.Verbose | Where-Object { $connWarning.Matches($_.Message) }).Count | Should -Be $expectedWarnings

            $pwsh.Dispose()

            return $session
        }
```

And use like this:
```powershell
  It 'Connection persistence impacted by changing SkipCertificateCheck' {
            $uri = Get-WebListenerUrl -Https
            # This first request will throw because the certificate is invalid
            $session = $null
            $session = RunCheckingPersistence -Uri $uri  -Command 'Invoke-WebRequest -Uri $uri -SkipCertificateCheck:$false' -CaptureSession
            # No change in setting
            $session = RunCheckingPersistence -Uri $uri  -Command 'Invoke-WebRequest -Uri $uri -SkipCertificateCheck:$false' -Session $session
            # Skipping cert check changes persistence
            $session = RunCheckingPersistence -Uri $uri  -Command 'Invoke-WebRequest -Uri $uri -SkipCertificateCheck' -Session $session -LostPersistence
            # Same settings won't lost persistence
            $session = RunCheckingPersistence -Uri $uri  -Command 'Invoke-WebRequest -Uri $uri -SkipCertificateCheck' -Session $session
            # Lose persistence due to changing cert check - this will also throw
            $session = RunCheckingPersistence -Uri $uri  -Command 'Invoke-WebRequest -Uri $uri -SkipCertificateCheck:$false' -Session $session -LostPersistence
        }
```
Before I commit all these changes, could you please give me some feedback on the approach. This took me a long time to get working because I couldn't just keep adding commands to the $pwsh item without it running all commands every time I ran it, and when I used the Clear commands as below between invocations of the same $pwsh, it stopped working for subsequent requests and gave me nothing in the stream.

```powershell
            $pwsh.Commands.Clear()
            $pwsh.Streams.ClearStreams()
```

Reply:
Perhaps we could makes the test more simple with test hook. See `InternalTestHooks` in Utils.cs.

Reply:
I'm not familiar with that, but it looks like it's used to control behaviour with flags while testing?

Are you suggesting I add a boolean to the static class there with something like:

```csharp
        //True if the last Invoke-WebRequest or Invoke-RestMethod had to reset the connection
        internal static bool LastWebRequestResetConnection;
```

And then gain access to the boolean after each request to validate?

Reply:
In the meantime, I have committed the tests using the Verbose stream approach.